src/test/java/com/github/lwhite1/tablesaw/TableTest.java

```diff
@@ -14,35 +22,61 @@
  */
 public class TableTest {
 
-  private Table table;
-  private FloatColumn column = new FloatColumn("f1");
-
-  @Before
-  public void setUp() throws Exception {
-    table = new Table("t");
-    table.addColumn(column);
-  }
-
-  @Test
-  public void testColumn() throws Exception {
-    Column column1 = table.column(0);
-    assertNotNull(column1);
-  }
-
-  @Test
-  public void testColumnCount() throws Exception {
-    assertEquals(0, new Table("t").columnCount());
-    assertEquals(1, table.columnCount());
-  }
-
-  @Test
-  public void testRowCount() throws Exception {
-    assertEquals(0, table.rowCount());
-    FloatColumn floatColumn = column;
-    floatColumn.add(2f);
-    assertEquals(1, table.rowCount());
-
-    floatColumn.add(2.2342f);
-    assertEquals(2, table.rowCount());
-  }
+    private Table table;
+    private FloatColumn column = new FloatColumn("f1");
+
+    @Before
+    public void setUp() throws Exception {
+        table = new Table("t");
+        table.addColumn(column);
+    }
+
+    @Test
+    public void testColumn() throws Exception {
+        Column column1 = table.column(0);
+        assertNotNull(column1);
+    }
+
+    @Test
+    public void testColumnCount() throws Exception {
+        assertEquals(0, new Table("t").columnCount());
+        assertEquals(1, table.columnCount());
+    }
+
+    @Test
+    public void testRowCount() throws Exception {
+        assertEquals(0, table.rowCount());
+        FloatColumn floatColumn = column;
+        floatColumn.add(2f);
+        assertEquals(1, table.rowCount());
+
+        floatColumn.add(2.2342f);
+        assertEquals(2, table.rowCount());
+    }
+
+    @Test
+    public void testGetRow() throws Exception {
+
+        Table table = TestData.SIMPLE_UNSORTED_DATA.getTable();
+        Path source = TestData.SIMPLE_UNSORTED_DATA.getSource();
+        List<String[]> rawData;
+        try (CSVReader csvReader = new CSVReader(Files.newBufferedReader(source, Charset.defaultCharset()))) {
+            rawData = csvReader.readAll();
+        }
+        // remove the first row from the raw data since tables don't keep column header row as part of their data
+        rawData.remove(0);
+        int maxRows = table.rowCount();
+        for (int rowIndex = 0; rowIndex < maxRows; rowIndex++) {
+            String row = table.getRow(rowIndex);
+            String expectedRow = getCSVfromListOfStrings(rawData.get(rowIndex));
+            assertEquals("Row " + rowIndex + " matches", expectedRow,row);
+        }
```

Comment:
I don't think you want a getRow() method in the Table class at all. It
should be a test utility method. It should use OpenCSV to construct "csv"
version of the table row, or it will fail on funky data.



On Sun, Jun 19, 2016 at 9:53 AM, Spencer Marks <notifications@github.com>
wrote:

> In src/test/java/com/github/lwhite1/tablesaw/TableTest.java
> <https://github.com/lwhite1/tablesaw/pull/19#discussion_r67617382>:
>
> > +    public void testGetRow() throws Exception {
> > +
> > +        Table table = TestData.SIMPLE_UNSORTED_DATA.getTable();
> > +        Path source = TestData.SIMPLE_UNSORTED_DATA.getSource();
> > +        List<String[]> rawData;
> > +        try (CSVReader csvReader = new CSVReader(Files.newBufferedReader(source, Charset.defaultCharset()))) {
> > +            rawData = csvReader.readAll();
> > +        }
> > +        // remove the first row from the raw data since tables don't keep column header row as part of their data
> > +        rawData.remove(0);
> > +        int maxRows = table.rowCount();
> > +        for (int rowIndex = 0; rowIndex < maxRows; rowIndex++) {
> > +            String row = table.getRow(rowIndex);
> > +            String expectedRow = getCSVfromListOfStrings(rawData.get(rowIndex));
> > +            assertEquals("Row " + rowIndex + " matches", expectedRow,row);
> > +        }
>
> In the test I had to make sure the getRow method worked. So I had to use
> the direct OpenCVS reader to get the "known" good data. (I could have
> gotten that data any number of ways, including just hard coding it, but
> this way seemed OK.)
>
> Then I compare it to the expected data which TableSaw provides.
>
> Does this make sense?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/lwhite1/tablesaw/pull/19/files/cd380a70fe6a53499b7d3ac576d609ae1dc77932#r67617382>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADRXgpS5phTI74c9aAKhSvrkMtW9ZOhzks5qNUnggaJpZM4I5ChM>
> .
>